solr/modules/monitor/src/test/org/apache/solr/monitor/MonitorSolrQueryTest.java

```diff
@@ -0,0 +1,612 @@
+      assertEquals(4, ((SolrDocumentList) response.getResponse().get("response")).size());
+    }
+  }
+
+  @Test
+  public void testDeleteByQueryId() throws Exception {
+    index(
+        id,
+        Integer.toString(0),
+        MonitorFields.MONITOR_QUERY,
+        "{!xmlparser}<DisjunctionMaxQuery><TermQuery fieldName=\"content0_s\">elevator</TermQuery><TermQuery fieldName=\"content1_s\">lift</TermQuery></DisjunctionMaxQuery>");
+    commit();
+    handle.clear();
+    handle.put("responseHeader", SKIP);
+    handle.put("response", SKIP);
+
+    Object[] params =
+        new Object[] {
+          CommonParams.SORT,
+          id + " desc",
+          CommonParams.JSON,
+          read("/monitor/dangling-test-single-doc-batch.json"),
+          CommonParams.QT,
+          "/reverseSearch",
+          WRITE_TO_DOC_LIST_KEY,
+          supportsWriteToDocList(),
+          QUERY_MATCH_TYPE_KEY,
+          "simple"
+        };
+
+    QueryResponse response = query(params);
+    if (supportsWriteToDocList()) {
+      assertEquals(1, ((SolrDocumentList) response.getResponse().get("response")).size());
+    }
+    validate(response, 0, List.of("0"));
+
+    del(MonitorFields.QUERY_ID + ":0");
+    commit();
+    response = query(CommonParams.Q, "*:*");
+    assertEquals(0, ((SolrDocumentList) response.getResponse().get("response")).size());
+  }
+
+  static Map<Object, Object> queryMatch(
+      String queryId,
+      String field1,
+      List<HighlightsMatch.Hit> firstFieldHits,
+      String field2,
+      List<HighlightsMatch.Hit> secondFieldHits) {
+    Map<Object, Object> queryMatch = new LinkedHashMap<>();
+    queryMatch.put(MonitorFields.QUERY_ID, queryId);
+    Map<Object, Object> matchHits = new LinkedHashMap<>();
+    var outHits1 = map(firstFieldHits);
+    if (!outHits1.isEmpty()) {
+      matchHits.put(field1, outHits1);
+    }
+    var outHits2 = map(secondFieldHits);
+    if (!outHits2.isEmpty()) {
+      matchHits.put(field2, outHits2);
+    }
+    queryMatch.put(MonitorConstants.HITS_KEY, matchHits);
+    return queryMatch;
+  }
+
+  static List<Object> map(List<HighlightsMatch.Hit> hits) {
+    List<Object> outHits = new ArrayList<>();
+    for (var hit : hits) {
+      Map<Object, Object> hit00 = new LinkedHashMap<>();
+      hit00.put("startPosition", hit.startPosition);
+      hit00.put("endPosition", hit.endPosition);
+      hit00.put("startOffset", hit.startOffset);
+      hit00.put("endOffset", hit.endOffset);
+      outHits.add(hit00);
+    }
+    return outHits;
+  }
+
+  void validate(QueryResponse response, int doc, Object expectedValue) {
+    var monitorQueries = monitorQueries(response, doc);
+    assertEquals(expectedValue, monitorQueries);
+  }
+
+  void validate(QueryResponse response, int doc, int query, Object expectedValue) {
+    var monitorQueries = monitorQueries(response, doc);
+    assertTrue(monitorQueries.size() > query);
+    assertEquals(expectedValue, monitorQueries.get(query));
+  }
```

Comment:
Hi @kotman12 - would you be comfortable adding me as a collaborator for your Solr fork repo? This would then, hopefully, allow sending of pull requests to the https://github.com/kotman12/solr/tree/solr-monitor branch for this pull request. I'd like to make some suggestions which might be easiest/clearest shared that way.

Reply:
@cpoerschke sure thing .. let me know if you got the invite, I've never used this feature before.

Reply:
Thanks! Yes, it worked i.e. I could send https://github.com/kotman12/solr/pull/1 now then.